gcpy/1D_raveller.py

```diff
@@ -0,0 +1,145 @@
+import argparse
+import hashlib
+import os.path
+
+import numpy as np
+import xarray as xr
+import pandas as pd
+
+from gcpy.grid import make_grid_CS
+
+def create_track_func(args):
+    nf = np.linspace(1, 6, 6)
+    Ydim = np.linspace(1, args.cs_res, args.cs_res)
+    Xdim = np.linspace(1, args.cs_res, args.cs_res)
+
+    grid, gridlist = make_grid_CS(args.cs_res)
+    #grid, gridlist = make_grid_SG(args.cs_res, *args.sg_params)
+
+    lon = xr.DataArray(grid['lon'] % 360, coords={'nf': nf, 'Ydim': Ydim, 'Xdim': Xdim}, dims=['nf', 'Ydim', 'Xdim'])
+    lon.values[lon.values > 180] -= 360
+    lat = xr.DataArray(grid['lat'], coords={'nf': nf, 'Ydim': Ydim, 'Xdim': Xdim}, dims=['nf', 'Ydim', 'Xdim'])
+
+    ds = xr.Dataset({'longitude': lon, 'latitude': lat})
+
+    #ds['time'] = (((ds['longitude'] / 15) + args.overpass_time) + (ds['latitude'] / 180) * args.vertical_scan_time) % 24
+    overpass_offset = ds['latitude']/90 * 24/args.orbits_per_day/4 * 60  # vary overpass time with latitude
+    if args.direction == 'ascending':
+        overpass_offset = -overpass_offset  # overpass delayed at high northern latitudes if ascending
+
+    overpass_time_timedelta_min = ds['longitude']/360 * 24 * 60 + overpass_offset
+
+    overhead_time = pd.to_datetime(args.overpass_time, format='%H:%M').time()
+    ds['time'] = (overhead_time.hour + overhead_time.minute/60 + overpass_time_timedelta_min/60) % 24
+
+    ds = ds.stack(track=['nf', 'Ydim', 'Xdim'])
+    ds = ds.sortby('time')
+
+    ds = ds.reset_index('track')
+    ds = ds.assign_coords({'track': ds.time}).drop('time').rename({'track': 'time'})
+
+    ds['longitude'].attrs['long_name'] = 'longitude'
+    ds['longitude'].attrs['units'] = 'degrees_east'
+
+    ds['latitude'].attrs['long_name'] = 'latitude'
+    ds['latitude'].attrs['units'] = 'degrees_north'
+
+    ds2 = ds.drop(['nf', 'Ydim', 'Xdim'])
+    ds2['nf'] = xr.DataArray(ds.nf.values, dims=['time'])
+    ds2['Ydim'] = xr.DataArray(ds.Ydim.values, dims=['time'])
+    ds2['Xdim'] = xr.DataArray(ds.Xdim.values, dims=['time'])
+
+    ds2['time'].attrs['long_name'] = 'time'
+    ds2['time'].attrs['units'] = 'hours since 1900-01-01 00:00:00'
```

Comment:
Note that 1900-01-01 is because we're using the "recycle" option for the 1D diagnostics which replace the date in the track file with the current simulation day.